// @uiw/codemirror-theme-abcdef/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.abcdef = void 0;

var _highlight = require("@lezer/highlight");

var _codemirrorThemes = require("@uiw/codemirror-themes");

/**
 * @name abcdef
 * @author codemirror.net
 * https://codemirror.net/5/theme/abcdef.css
 */
var abcdef = (0, _codemirrorThemes.createTheme)({
  theme: 'dark',
  settings: {
    background: '#0f0f0f',
    foreground: '#defdef',
    caret: '#00FF00',
    selection: '#515151',
    selectionMatch: '#515151',
    gutterBackground: '#555',
    gutterForeground: '#FFFFFF',
    lineHighlight: '#314151'
  },
  styles: [{
    tag: _highlight.tags.keyword,
    color: 'darkgoldenrod',
    fontWeight: 'bold'
  }, {
    tag: _highlight.tags.atom,
    color: '#77F'
  }, {
    tag: _highlight.tags.comment,
    color: '#7a7b7c',
    fontStyle: 'italic'
  }, {
    tag: _highlight.tags.number,
    color: 'violet'
  }, {
    tag: _highlight.tags.definition(_highlight.tags.variableName),
    color: '#fffabc'
  }, {
    tag: _highlight.tags.variableName,
    color: '#abcdef'
  }, {
    tag: _highlight.tags["function"](_highlight.tags.variableName),
    color: '#fffabc'
  }, {
    tag: _highlight.tags.typeName,
    color: '#FFDD44'
  }, {
    tag: _highlight.tags.tagName,
    color: '#def'
  }, {
    tag: _highlight.tags.string,
    color: '#2b4'
  }, {
    tag: _highlight.tags.meta,
    color: '#C9F'
  }, // { tag: t.qualifier, color: '#FFF700' },
  // { tag: t.builtin, color: '#30aabc' },
  {
    tag: _highlight.tags.bracket,
    color: '#8a8a8a'
  }, {
    tag: _highlight.tags.attributeName,
    color: '#DDFF00'
  }, {
    tag: _highlight.tags.heading,
    color: 'aquamarine',
    fontWeight: 'bold'
  }, {
    tag: _highlight.tags.link,
    color: 'blueviolet',
    fontWeight: 'bold'
  }]
});
exports.abcdef = abcdef;
